tests/test_main.py

```diff
@@ -0,0 +1,178 @@
+import pytest
+from fastapi.testclient import TestClient
+from raise_search.api.main import app
+from raise_search.api.utils import get_opensearch_client
+
+
+@pytest.fixture
+def test_client():
+    return TestClient(app)
+
+
+def create_mock_response():
+    mock_response = {
+        "took": 5,
+        "timed_out": False,
+        "_shards": {"total": 5, "successful": 5, "skipped": 0, "failed": 0},
+        "hits": {
+            "total": {"value": 1, "relation": "eq"},
+            "hits": [
+                {
+                    "_index": "content-1",
+                    "_id": "1",
+                    "_score": 0.5,
+                    "_source": {
+                        "content_id": "1",
+                        "section": "section1",
+                        "activity_name": "Activity 1",
+                        "lesson_page": "page1",
+                        "lesson_page_type": "type1",
+                        "teacher_only": True,
+                        "visible_content": ["content"],
+                    },
+                    "highlight": {
+                        "visible_content": ["text"],
+                        "lesson_page": ["page"],
+                        "activity_name": ["activity"],
+                    },
+                },
+            ],
+            "max_score": 1.0,
+        },
+    }
+    return mock_response
+
+
+@pytest.fixture
+def mock_opensearch_client(mocker):
+    mock_client = mocker.MagicMock()
+    mock_response = create_mock_response()
+    mock_client.search.return_value = mock_response
+    return mock_client
+
+
+def test_search_v1(mock_opensearch_client, test_client):
+    client = TestClient(app)
+
+    app.dependency_overrides[get_opensearch_client] = lambda: mock_opensearch_client
+
+    test_query = "test query"
+    test_version = "1"
+
+    response = client.get(f"/v1/search?q={test_query}&version={test_version}")
+
+    assert response.status_code == 200
```

Comment:
Should we add assertions on the response structure? (perhaps in just this test case would be sufficient if so)